src/pip/_internal/cli/cmdoptions.py

```diff
@@ -794,6 +794,33 @@ def _handle_no_use_pep517(
     help=SUPPRESS_HELP,
 )
 
+
+def _handle_config_settings(
+    option: Option, opt_str: str, value: str, parser: OptionParser
+) -> None:
+    key, sep, val = value.partition("=")
```

Comment:
Should we enforce that value does not have `=` unless quoted, or something like that? How does specifying this in a configuration file behave?

Reply:
Why? What's wrong with `--config-settings EXPR=x=2` (apart from probably being pointless and definitely being silly)?

I assume that setting it in a config file (or environment variable) would work just as for any other option (i.e., it would apply to all invocations). Was there a particular thing you thought might go wrong?

One thing is that if a setting FOO is set in a config file, it can't be unset on the command line. I originally considered `--config-settings FOO=` to unset FOO, but @uranusjr didn't think it was worth it (and honestly, why unset it rather than set it to an empty string?) So I'm going with YAGNI for now on that.

Reply:
> that value does not have `=` unless quoted

This is CLI parsing, and since quotes are eagerly stripped by the shell, the command line would be come something like this if we expect them here:

```bash
# Simply --config-settings EXPR="x=2" would become EXPR=x=2 in the program!
pip install foo --config-settings 'EXPR="x=2"'
```

which is extremely unintuitive and not worth it. If we want to enforce quotes (which I also _don’t_ think is a good idea), it needs to be in the requirement file parser, not the CLI argument parser.

Reply:
> Why? What's wrong with --config-settings EXPR=x=2 (apart from probably being pointless and definitely being silly)?

My thought process was that its pointless, and it might be the result of user error?

Reply:
> since quotes are eagerly stripped by the shell

That's a fair point, and I'm basically sure that doing this would be a bad idea. :)

Reply:
I can't actually think what a user might mean that would make parsing it as {"EXPR": "x=2"} a problem. Users could just as easily mis-spell the key name, and we're not trying to protect them against that.

If we get users complaining that they made errors like you suggest, let's add a validation check, or a more featureful parser, then. For now, we literally only have one known case where config settings are needed, so I think it's probably premature to worry about this level of detail.